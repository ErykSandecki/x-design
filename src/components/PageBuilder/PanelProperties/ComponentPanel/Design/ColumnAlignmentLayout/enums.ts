export enum PopoverItem {
  fixed = 0,
}
