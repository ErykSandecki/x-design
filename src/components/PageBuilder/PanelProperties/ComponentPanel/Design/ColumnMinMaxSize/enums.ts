export enum PopoverItem {
  currentValue = 0,
  auto = 1,
  unit = 2,
  removeScore = 3,
}
