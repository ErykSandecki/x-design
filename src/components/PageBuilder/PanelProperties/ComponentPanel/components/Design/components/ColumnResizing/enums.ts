export enum PopoverItem {
  auto = 1,
  fixed = 0,
  maxScore = 4,
  minScore = 3,
  unit = 2,
}
