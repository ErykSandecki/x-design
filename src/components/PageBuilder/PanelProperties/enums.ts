export enum Tab {
  design = 'design',
  prototype = 'prototype',
}
