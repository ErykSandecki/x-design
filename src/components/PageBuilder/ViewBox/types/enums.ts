export enum ElementStickWallPosition {
  bottom = 'bottom',
  left = 'left',
  right = 'right',
  top = 'top',
}
