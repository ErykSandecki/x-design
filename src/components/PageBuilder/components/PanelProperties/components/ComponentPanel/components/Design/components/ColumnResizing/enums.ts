export enum PopoverItem {
  auto = 1,
  fixed = 0,
  unit = 2,
}
