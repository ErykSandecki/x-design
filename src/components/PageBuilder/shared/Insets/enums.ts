export enum InsetMode {
  individual = 'individual',
  merged = 'merged',
}
