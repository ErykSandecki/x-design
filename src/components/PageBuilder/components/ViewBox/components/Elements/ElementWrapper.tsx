import { FC, memo, ReactNode } from 'react';
import { useSelector } from 'react-redux';

// components
import DropAnchors from './components/DropAnchors/DropAnchors';

// store
import { elementAttributeSelectorCreator } from 'store/pageBuilder/selectors';

// types
import { MouseMode, TElement } from 'types';

export type TElementWrapperProps = {
  children: ReactNode;
  id: TElement['id'];
  index: number;
  isSelected: boolean;
  mouseMode: MouseMode;
  parentId: TElement['parentId'];
};

const ElementWrapper: FC<TElementWrapperProps> = ({
  children,
  id,
  index,
  isSelected,
  mouseMode,
  parentId,
}) => {
  const layoutParent = useSelector(
    elementAttributeSelectorCreator('layout', parentId),
  );
  const position = useSelector(elementAttributeSelectorCreator('position', id));

  if (!isSelected && position === 'relative') {
    return (
      <DropAnchors
        id={id}
        index={index}
        mouseMode={mouseMode}
        layoutParent={layoutParent}
        parentId={parentId}
      >
        {children}
      </DropAnchors>
    );
  }

  return children;
};

export default memo(ElementWrapper);
