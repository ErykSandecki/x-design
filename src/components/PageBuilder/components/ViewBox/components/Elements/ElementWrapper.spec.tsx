import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';

// components
import ElementWrapper from './ElementWrapper';

// types
import { MouseMode } from 'types/enums/mouseMode';

jest.mock('./components/DropAnchors/DropAnchors', () => ({ children }) => (
  <div className="drop-anchors">{children}</div>
));

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('ElementWrapper snapshots', () => {
  it('should render ElementWrapper', () => {
    // mock
    (useSelector as unknown as jest.Mock).mockImplementation(() => 'absolute');

    // before
    const { asFragment } = render(
      <ElementWrapper
        id="-1"
        isSelected={false}
        mouseMode={MouseMode.default}
        parentId="-1"
      >
        children
      </ElementWrapper>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with drop anchors', () => {
    // mock
    (useSelector as unknown as jest.Mock).mockImplementation(() => 'relative');

    // before
    const { asFragment } = render(
      <ElementWrapper
        id="-1"
        isSelected={false}
        mouseMode={MouseMode.default}
        parentId="-1"
      >
        children
      </ElementWrapper>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
