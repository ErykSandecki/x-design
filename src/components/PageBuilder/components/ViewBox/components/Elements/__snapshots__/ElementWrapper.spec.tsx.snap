// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementWrapper snapshots should render ElementWrapper 1`] = `
<DocumentFragment>
  children
</DocumentFragment>
`;

exports[`ElementWrapper snapshots should render with drop anchors 1`] = `
<DocumentFragment>
  <div
    class="drop-anchors"
  >
    children
  </div>
</DocumentFragment>
`;
